File: clever-record-keeper-main/server/checkMongo.js
import { MongoClient } from 'mongodb';

const MONGO_URI = process.env.MONGO_URI || 'mongodb://localhost:27017';
const DB_NAME = process.env.MONGO_DB || 'clever_record_keeper';

(async () => {
  const client = new MongoClient(MONGO_URI, { serverSelectionTimeoutMS: 5000 });
  try {
    console.log('Connecting to', MONGO_URI, 'db', DB_NAME);
    await client.connect();
    const db = client.db(DB_NAME);
    const col = db.collection('students');
    const count = await col.countDocuments();
    console.log('students.count =', count);
    const first = await col.find({}).limit(1).toArray();
    console.log('first doc sample:', JSON.stringify(first[0], null, 2));
  } catch (err) {
    console.error('Connection / query failed:', err.message || err);
    process.exitCode = 2;
  } finally {
    await client.close();
  }
})();

File: clever-record-keeper-main/server/check_mongo.py
from pymongo import MongoClient
from bson.json_util import dumps
import sys

# Configuration
MONGO_URI = "mongodb://localhost:27017"
DB_NAME = "clever_record_keeper"
COLLECTION_NAME = "students"
SAMPLE_LIMIT = 5  # how many sample docs to print

def main():
    try:
        client = MongoClient(MONGO_URI, serverSelectionTimeoutMS=5000)
        # Force connection / server selection
        client.admin.command('ping')
    except Exception as e:
        print(f"ERROR: Unable to connect to MongoDB at {MONGO_URI}: {e}", file=sys.stderr)
        sys.exit(2)

    db = client[DB_NAME]
    col = db[COLLECTION_NAME]

    try:
        count = col.count_documents({})
        print(f"Connected to {MONGO_URI} -> DB: {DB_NAME} -> Collection: {COLLECTION_NAME}")
        print(f"Document count: {count}")
        if count == 0:
            print("Collection is empty.")
        else:
            print(f"\nShowing up to {SAMPLE_LIMIT} sample documents:\n")
            cursor = col.find({}).limit(SAMPLE_LIMIT)
            for i, doc in enumerate(cursor, start=1):
                # dumps converts BSON types (ObjectId, Date) to JSON-friendly format
                print(f"--- Document {i} ---")
                print(dumps(doc, indent=2))
                print()
    except Exception as e:
        print("ERROR while querying the collection:", e, file=sys.stderr)
        sys.exit(3)
    finally:
        client.close()

if __name__ == '__main__':
    main()